packages/nimble-components/src/table/models/update-tracker.ts

```diff
@@ -0,0 +1,143 @@
+
+    public get updateColumnSort(): boolean {
+        return this.requiredUpdates.columnSort;
+    }
+
+    public get updateColumnDefinition(): boolean {
+        return this.requiredUpdates.columnDefinition;
+    }
+
+    public get updateActionMenuSlots(): boolean {
+        return this.requiredUpdates.actionMenuSlots;
+    }
+
+    public get requiresTanStackUpdate(): boolean {
+        return (
+            this.requiredUpdates.rowIds
+            || this.requiredUpdates.columnSort
+            || this.requiredUpdates.columnDefinition
+        );
+    }
+
+    public get requiresTanStackDataReset(): boolean {
+        return (
+            this.requiredUpdates.rowIds || this.requiredUpdates.columnDefinition
+        );
+    }
+
+    public trackAllStateChanged(): void {
+        this.setAllKeys(true);
+        this.queueUpdate();
+    }
+
+    public trackColumnPropertyChanged(changedColumnProperty: string): void {
+        if (this.isSameProperty(changedColumnProperty, 'columnId')) {
+            this.requiredUpdates.columnIds = true;
+        } else if (
+            this.isSameProperty(
+                changedColumnProperty,
+                'operandDataRecordFieldName'
+            )
+            || this.isSameProperty(changedColumnProperty, 'sortOperation')
+        ) {
+            this.requiredUpdates.columnDefinition = true;
+        } else if (
+            this.isSameProperty(changedColumnProperty, 'sortIndex')
+            || this.isSameProperty(changedColumnProperty, 'sortDirection')
+        ) {
+            this.requiredUpdates.columnSort = true;
+        } else if (
+            this.isSameProperty(changedColumnProperty, 'actionMenuSlot')
+        ) {
+            this.requiredUpdates.actionMenuSlots = true;
+        }
+
+        this.queueUpdate();
+    }
+
+    public trackColumnInstancesChanged(): void {
+        this.requiredUpdates.columnIds = true;
+        this.requiredUpdates.columnDefinition = true;
+        this.requiredUpdates.columnSort = true;
+        this.requiredUpdates.actionMenuSlots = true;
+
+        this.queueUpdate();
+    }
+
+    public trackIdFieldNameChanged(): void {
+        this.requiredUpdates.rowIds = true;
+        this.queueUpdate();
+    }
+
+    private setAllKeys(value: boolean): void {
+        Object.keys(this.requiredUpdates).forEach(key => {
+            this.requiredUpdates[key] = value;
+        });
+    }
+
+    private queueUpdate(): void {
+        if (!this.table.$fastController.isConnected) {
+            return;
+        }
+
+        if (!this.updateQueued) {
+            this.updateQueued = true;
+            DOM.queueUpdate(() => {
+                this.table.update();
+                this.setAllKeys(false);
+                this.updateQueued = false;
+            });
+        }
+    }
+
+    private isSameProperty(
+        changedProperty: string,
+        columnProperty: keyof TableColumn
+    ): boolean {
+        return changedProperty === columnProperty;
+    }
```

Comment:
nit / pref: maybe make it a function outside of the class in the file (so can avoid the `this` prefix since it doesn't need instance state) and a name like `isColumnProperty`. Can also take an args for comparison:

ex:
```ts
const isColumnProperty = (
    changedProperty: string,
    ...args: (keyof TableColumn)[]
): boolean => {
    for (const arg of args) {
        if (changedProperty === arg) {
            return true;
        }
    }
    return false;
};
```

before:
```ts
if (this.isSameProperty(changedColumnProperty, 'operandDataRecordFieldName')
    || this.isSameProperty(changedColumnProperty, 'sortOperation'))
```
after:
```ts
if (isColumnProperty(changedColumnProperty, 'operandDataRecordFieldName', 'sortOperation'))
```